contracts/Proxy.sol

```diff
@@ -31,4 +32,20 @@ contract Proxy {
             default { return(0, returndatasize()) }
         }
     }
+
+    function implementation()
+        public
+        view
+        returns (address)
+    {
+        return masterCopy;
+    }
+
+    function proxyType()
+        public
+        pure
+        returns (uint256)
+    {
+        return 2;
```

Comment:
Not sure if these methods should be declared here or in the master copy

Reply:
I think the intention was to implement them in `Proxy.sol.`